lib/graph.ml

```diff
@@ -0,0 +1,90 @@
+type color = Red | Blue | Green | Yellow
+type position = int * int
```

Comment:
Maybe give them names. Even though I find this representation very odd what do you mean by int Int. The row and the column? But what is the first row and the first column in a torus:/.
I personally thought more about having a list of neighbors. Maybe discuss it with Mr. Rabie?

Reply:
for positions, this is how it's normally represented: x, y with x the column number and y the row number, it's common in programs, so I don't see the point of specifying it